assets/src/hooks/useFetchTimepoints.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react"
+import { fetchTimepointsForRoute } from "../api"
+import { RouteId, Timepoint, TimepointsByRouteId } from "../skate"
+import {
+  Dispatch,
+  setLoadingTimepointsForRoute,
+  setTimepointsForRoute,
+} from "../state"
+
+export const useFetchTimepoints = (
+  selectedRouteIds: RouteId[],
+  timepointsByRouteId: TimepointsByRouteId,
+  dispatch: Dispatch
+): void => {
+  useEffect(() => {
+    selectedRouteIds.forEach((routeId: RouteId) => {
+      if (!(routeId in timepointsByRouteId)) {
+        dispatch(setLoadingTimepointsForRoute(routeId))
+
+        fetchTimepointsForRoute(routeId).then((newTimepoints: Timepoint[]) =>
```

Comment:
why is fetching the data something defined here in the effect, rather than something handled by an event in `dispatch`?

Reply:
The reducer has to be pure, so it can't trigger the fetch.

Reply:
What happens when there's an error?

Reply:
Right now, we don't handle that case. If we want to handle it in the future, we'd add a `SetFetchErrorAction`, and then add a case to the `.then(`, which would call dispatch with that action.

Reply:
Can we log something to `console.error` so that it's not failing silently?

Reply:
`fetchTimepointsForRoute` does have a `catch` block that's logging the error to `console.error`. In general our error handling and UI needs a bunch of thought - I'm going to make a ticket to at least have a place holder.

Reply:
Adapting an existing ticket: https://app.asana.com/0/1112935048846093/1117662445344607. Please feel free to add anything there that would be helpful.